src/sign.rs

```diff
@@ -547,78 +516,255 @@ impl CoseSign1 {
 
 #[cfg(test)]
 mod tests {
-    use super::*;
-    use openssl::pkey::PKey;
-
-    // Public domain work: Pride and Prejudice by Jane Austen, taken from https://www.gutenberg.org/files/1342/1342.txt
-    const TEXT: &[u8] = b"It is a truth universally acknowledged, that a single man in possession of a good fortune, must be in want of a wife.";
-
-    #[test]
-    fn map_serialization() {
-        // Empty map
-        let map: HeaderMap = HeaderMap::new();
-        assert_eq!(map_to_empty_or_serialized(&map).unwrap(), []);
-
-        // Checks that the body_protected field will be serialized correctly
-        let map: HeaderMap = SignatureAlgorithm::ES256.into();
-        assert_eq!(
-            map_to_empty_or_serialized(&map).unwrap(),
-            [0xa1, 0x01, 0x26]
-        );
-
-        let map: HeaderMap = SignatureAlgorithm::ES384.into();
-        assert_eq!(
-            map_to_empty_or_serialized(&map).unwrap(),
-            [0xa1, 0x01, 0x38, 0x22]
-        );
-
-        let map: HeaderMap = SignatureAlgorithm::ES512.into();
-        assert_eq!(
-            map_to_empty_or_serialized(&map).unwrap(),
-            [0xa1, 0x01, 0x38, 0x23]
-        );
-    }
+    #[cfg(feature = "key_openssl_pkey")]
+    mod generic {
+        use crate::sign::*;
+        use openssl::pkey::{PKey, Private, Public};
+
+        // Public domain work: Pride and Prejudice by Jane Austen, taken from https://www.gutenberg.org/files/1342/1342.txt
+        const TEXT: &[u8] = b"It is a truth universally acknowledged, that a single man in possession of a good fortune, must be in want of a wife.";
+
+        #[test]
+        fn map_serialization() {
```

Comment:
This test doesn't depend on openssl, so it should be moved outside of `mod generic`.